stl/inc/memory

```diff
@@ -936,6 +935,20 @@ namespace ranges {
 #endif // _HAS_CXX17
 
 
+template <class _PtrTy>
+_NODISCARD void* _Voidify_unfancy(_PtrTy _Ptr) noexcept {
+    if constexpr (is_pointer_v<_PtrTy>) {
+        return const_cast<void*>(static_cast<const volatile void*>(_Ptr));
+    } else {
+        return const_cast<void*>(static_cast<const volatile void*>(_STD addressof(*_Ptr)));
+    }
+}
+
+template <class _Ty, class... _Types>
+void _Construct_cv_in_place(_Ty& _Obj, _Types&&... _Args) noexcept(is_nothrow_constructible_v<_Ty, _Types...>) {

```

Comment:
I'd prefer to remove these and mostly use the same implementation as `optional`:

- `raw_storage_iterator`, as a deprecated thing, should do the `const_cast<static_cast>` thing itself
- `_Ref_count_obj2` should use `_Wrap<remove_cv_t<_Ty>>`
- `_Ref_count_[un]bounded_array`,  should use `_Wrap<remove_cv_t<_Ty>>` and `_Uninitialized_*_multidimensional` should remove the `_Voidify`